src/renderers/dom/shared/ReactDOMComponent.js

```diff
@@ -139,6 +149,16 @@ function assertValidProps(component, props) {
     'not a string. For example, style={{marginRight: spacing + \'em\'}} when ' +
     'using JSX.'
   );
+  if (__DEV__) {
+    Object.keys(props).forEach(function(propName) {
+      warning(
+        !eventTypes.hasOwnProperty(propName),
```

Comment:
and would fail with this edge-case:

```javascript
var eventTypes = {}
var eventType = eventTypes["constructor"]
!eventTypes || eventType === "constructor" // false
```